openwrt-openwisp-monitoring/files/monitoring.agent

```diff
@@ -167,38 +161,52 @@ send_data() {
                    -p daemon.err
           fi
 
-          [ "$verbose_mode" = "0" ] && logger -s "Run with verbose mode to find more." \
-                                              -t openwisp_monitoring \
-                                              -p daemon.err
+          [ "$verbose_mode" -eq "0" ] && logger -s "Run with verbose mode to find more." \
+                                                -t openwisp_monitoring \
+                                                -p daemon.err
           break
         fi
-        #send data
+        # send data
+        data=$(cat "$file")
         response_code=$($CURL_COMMAND -H "Content-Type: application/json" \
           -d "$data" \
           -v "$url")
         if [ "$response_code" = "200" ]; then
-          [ "$verbose_mode" = "1" ] && logger -s "Data sent successfully." \
-                                              -p daemon.info
-          #Removed saved data
+          [ "$verbose_mode" -eq "1" ] && logger -s "Data sent successfully." \
+                                                -p daemon.info
+          #remove saved data 
           rm "$file"
           break
         else
-          TIME=$((BASE_SECONDS * 2 **(FAILURES-1)))
-          [ "$verbose_mode" = "1" ] && logger -s "Data not sent successfully. Retrying after $TIME seconds" \
-                                              -p daemon.err
+          TIME=$((BASE_SECONDS * 2 **(failures)))
+          [ "$verbose_mode" -eq "1" ] && logger -s "Data not sent successfully. Retrying in $TIME seconds" \
+                                                -p daemon.err
+          failures=$((failures+1))
           sleep "$TIME"
         fi
       done
     done
   done
 }
 
-if [ "$COLLECT" -eq "1" ]; then
+if [ -z "$MODE" ]; then
+  logger -s "missing required --mode option" \
+         -t openwisp_monitoring \
+         -p daemon.err
+  exit 1
+fi
+
+if [ "$MODE" = "collect" ]; then
   MONITORED_INTERFACES=${MONITORED_INTERFACES:-*}
+  # remove double quotes from interfaces 
+  MONITORED_INTERFACES=$(echo "$MONITORED_INTERFACES" | tr -d '"')
   save_data
-else
+elif [ "$MODE" = "send" ]; then
   VERIFY_SSL=${VERIFY_SSL:-0}
   set_url_and_curl && send_data
+else
+  logger -s "Please specify correct mode" \
```

Comment:
instead of logging an error, I suggest just echoing to standard error a message like "The supplied mode is invalid, only collect and send are accepted."